wordpress_org_api/src/plugin_directory.rs

```diff
@@ -105,3 +121,202 @@ pub struct Icons {
+        assert_eq!(plugin_with_variant_types.parsed.requires_php, "");
+
+        assert!(plugin_with_expected_types.raw_json["requires_php"].is_string());
+        assert!(!plugin_with_expected_types.parsed.requires_php.is_empty());
+    }
+
+    #[rstest]
+    fn test_plugin_with_different_types_of_upgrade_notice(
+        plugin_with_variant_types: Plugin,
+        plugin_with_expected_types: Plugin,
+    ) {
+        assert_eq!(
+            plugin_with_variant_types.raw_json["upgrade_notice"].as_array(),
+            Some(vec![]).as_ref()
+        );
+        assert_eq!(
+            plugin_with_variant_types.parsed.upgrade_notice,
+            HashMap::new()
+        );
+
+        assert!(plugin_with_expected_types.raw_json["upgrade_notice"].is_object());
+        assert!(!plugin_with_expected_types.parsed.upgrade_notice.is_empty());
+    }
+
+    #[rstest]
+    fn test_plugin_with_different_types_of_tags(
+        plugin_with_variant_types: Plugin,
+        plugin_with_expected_types: Plugin,
+    ) {
+        assert_eq!(
+            plugin_with_variant_types.raw_json["tags"].as_array(),
+            Some(vec![]).as_ref()
+        );
+        assert_eq!(plugin_with_variant_types.parsed.tags, HashMap::new());
+
+        assert!(plugin_with_expected_types.raw_json["tags"].is_object());
+        assert!(!plugin_with_expected_types.parsed.tags.is_empty());
+    }
+
+    #[rstest]
+    fn test_plugin_with_different_types_of_versions(
+        plugin_with_variant_types: Plugin,
+        plugin_with_expected_types: Plugin,
+    ) {
+        assert_eq!(
+            plugin_with_variant_types.raw_json["versions"].as_array(),
+            Some(vec![]).as_ref()
+        );
+        assert_eq!(plugin_with_variant_types.parsed.versions, HashMap::new());
+
+        assert!(plugin_with_expected_types.raw_json["versions"].is_object());
+        assert!(!plugin_with_expected_types.parsed.versions.is_empty());
+    }
+
+    #[rstest]
+    fn test_plugin_with_different_types_of_business_model(
+        plugin_with_variant_types: Plugin,
+        plugin_with_expected_types: Plugin,
+    ) {
+        assert_eq!(plugin_with_variant_types.raw_json["business_model"], false);
+        assert_eq!(plugin_with_variant_types.parsed.business_model, "");
+
+        assert!(plugin_with_expected_types.raw_json["business_model"].is_string());
+        assert!(!plugin_with_expected_types.parsed.business_model.is_empty());
+    }
+
+    #[rstest]
+    fn test_plugin_with_different_types_of_banners(
+        plugin_with_variant_types: Plugin,
+        plugin_with_expected_types: Plugin,
+    ) {
+        assert_eq!(
+            plugin_with_variant_types.raw_json["banners"].as_array(),
+            Some(vec![]).as_ref()
+        );
+        assert_eq!(plugin_with_variant_types.parsed.banners, Banners::default());
+
+        assert!(plugin_with_expected_types.raw_json["banners"].is_object());
+    }
+
+    fn test_plugin_query_result() {
```

Comment:
```suggestion
    #[test]
    fn test_plugin_query_result() {
```

This test seems to fail with:
> Failed to parse JSON: Some(Error("missing field `preview_link`", line: 36, column: 5))